### PR TITLE
[Makefile] Add development targets!

### PR DESCRIPTION
**Pull request by ayushkovskiy**
- `make develop`
- `make connect-develop`
- `make remote-port-forward`
- `make kill-develop`

### CODE REVIEW

File: {{cookiecutter.project_slug}}/Makefile [hunk 1]
```diff
@@ -116,6 +120,54 @@ clean: clean-code clean-data clean-notebooks
...
+		--name $(DEVELOP_JOB) \
+		--preset cpu-small \
+		--detach \
+		--volume $(DATA_DIR_STORAGE):$(PROJECT_PATH_ENV)/$(DATA_DIR):ro \
+		--volume $(PROJECT_PATH_STORAGE)/$(CODE_DIR):$(PROJECT_PATH_ENV)/$(CODE_DIR):rw \
+		--volume $(PROJECT_PATH_STORAGE)/$(RESULTS_DIR):$(PROJECT_PATH_ENV)/$(RESULTS_DIR):rw \
+		--env EXPOSE_SSH=yes \
+		--env JOB_LIFETIME=1d \
+		$(CUSTOM_ENV_NAME) \
+		"sleep infinity"
```
Comment (ayushkovskiy) [inline {{cookiecutter.project_slug}}/Makefile]: + exec

File: {{cookiecutter.project_slug}}/Makefile [hunk 2]
```diff
@@ -122,8 +126,56 @@ clean-all: clean-code clean-data clean-notebooks  ### Delete code, data, and not
...
+
+.PHONY: logs-develop
+logs-develop:  ### Connect to the remote shell running on the development job
+	$(NEURO) logs $(DEVELOP_JOB)
+
+    .PHONY: remote-debug-develop
+remote-debug-develop:  ### Forward SSH port to localhost for using PyCharm Pro remote debugging
+	@test ${REMOTE_DEBUG_PORT} || { echo 'Please set up env var REMOTE_DEBUG_PORT'; false; }
+	@echo
+	@echo 'Enabling remote debugging with PyCharm Pro:'
```
Comment (atselousov) [inline {{cookiecutter.project_slug}}/Makefile]: Maybe use vscode? At least it's free.

Comment (ayushkovskiy) [reply {{cookiecutter.project_slug}}/Makefile]: OK let's move this to documentation

File: {{cookiecutter.project_slug}}/Makefile [hunk 3]
```diff
@@ -122,6 +126,38 @@ clean-all: clean-code clean-data clean-notebooks  ### Delete code, data, and not
 
 ##### JOBS #####
 
+.PHONY: develop
+develop: upload-code upload-notebooks  ### Run a development job
```
Comment (atselousov) [inline {{cookiecutter.project_slug}}/Makefile]: Have we made any decision about hidden code uploading before a job starting?

Comment (ayushkovskiy) [reply {{cookiecutter.project_slug}}/Makefile]: cc @mariyadavydova @alekseysh 
We need to make a decision!

I propose: now merge with dependencies, and then, if we decide to solve https://github.com/neuromation/cookiecutter-neuro-project/issues/118,  we can get rid of all of them at once.

File: {{cookiecutter.project_slug}}/Makefile (hunk 2 above)
Comment (ayushkovskiy) [reply {{cookiecutter.project_slug}}/Makefile]: if the setup is the same there, it's easy. Instructions will be on neu.ro docs

File: {{cookiecutter.project_slug}}/Makefile [hunk 4]
```diff
@@ -44,6 +45,9 @@ HTTP_AUTH?=--http-auth
 # TRAINING_COMMAND="bash -c 'cd $(PROJECT_PATH_ENV) && python -u $(CODE_DIR)/train.py --data $(DATA_DIR)'"
 TRAINING_COMMAND?='echo "Replace this placeholder with a training script execution"'
 
+# Port for remote debugging via PyCharm Pro
```
Comment (mariyadavydova) [inline {{cookiecutter.project_slug}}/Makefile]: Let's remove PyCharm from here. We'll provide information in docs.

File: {{cookiecutter.project_slug}}/Makefile (hunk 3 above)
Comment (mariyadavydova) [reply {{cookiecutter.project_slug}}/Makefile]: ATM I agree with @ayushkovskiy. There are too many open conflicting PRs in this project now, let's resolve them all and get back to the dependencies story.

File: {{cookiecutter.project_slug}}/Makefile [hunk 5]
```diff
@@ -122,6 +126,38 @@ clean-all: clean-code clean-data clean-notebooks  ### Delete code, data, and not
...
+.PHONY: connect-develop
+connect-develop:  ### Connect to the remote shell running on the development job
+	$(NEURO) exec --no-key-check $(DEVELOP_JOB) bash
+
+.PHONY: logs-develop
+logs-develop:  ### Connect to the remote shell running on the development job
+	$(NEURO) logs $(DEVELOP_JOB)
+
+.PHONY: port-forward-develop
+port-forward-develop:  ### Forward SSH port to localhost for using PyCharm Pro remote debugging
```
Comment (mariyadavydova) [inline {{cookiecutter.project_slug}}/Makefile]: Again, let's remove PyCharm from here.

File: {{cookiecutter.project_slug}}/Makefile [hunk 6]
```diff
@@ -116,6 +120,54 @@ clean: clean-code clean-data clean-notebooks
...
+
+.PHONY: connect-develop
+connect-develop:  ### Connect to the remote shell running on the development job
+	$(NEURO) exec --no-tty --no-key-check $(DEVELOP_JOB) bash
+
+.PHONY: logs-develop
+logs-develop:  ### Connect to the remote shell running on the development job
+	$(NEURO) logs $(DEVELOP_JOB)
+
+    .PHONY: remote-debug-develop
```
[the remark this answers is not in the document]
Comment (ayushkovskiy) [reply {{cookiecutter.project_slug}}/Makefile]: outdated :)

File: {{cookiecutter.project_slug}}/Makefile (hunk 4 above)
Comment (ayushkovskiy) [reply {{cookiecutter.project_slug}}/Makefile]: ok thanks

**Pull request merged**